tests/perf/_perf_utils.py

```diff
@@ -0,0 +1,217 @@
+    results = []
+    for metric in HEADER:
+        results.append(res.get(metric, -1))
+    return results
+
+
+def convert_to_csv(filename, results, workers):
+    with open(filename, "w") as file:
+        writer = csv.writer(file)
+        writer.writerow(HEADER)
+        # Benchmarking main script uses Multiprocessing Pool.map(),
+        # thus results is structured as List[List[Dict[str, any]]].
+        for result in results:
+            for row in result:
+                writer.writerow(results_to_csv(row))
+
+
+def convert_to_cloud_monitoring(bucket_name, results, workers):
+    # Benchmarking main script uses Multiprocessing Pool.map(),
+    # thus results is structured as List[List[Dict[str, any]]].
+    for result in results:
+        for res in result:
+            range_read_size = res.get("RangeReadSize", 0)
+            object_size = res.get("ObjectSize")
+            elapsed_time_us = res.get("ElapsedTimeUs")
+            status = res.get("Status").pop()  # convert ["OK"] --> "OK"
+
+            # Handle range reads and calculate throughput using range_read_size.
+            if range_read_size > 0:
+                size = range_read_size
+            else:
+                size = object_size
+
+            # If size is greater than the defined threshold, report in MiB/s, otherwise report in KiB/s.
+            if size >= SSB_SIZE_THRESHOLD_BYTES:
+                throughput = (size / 1024 / 1024) / (elapsed_time_us / 1_000_000)
+            else:
+                throughput = (size / 1024) / (elapsed_time_us / 1_000_000)
+
+            cloud_monitoring_output = (
+                "throughput{"
+                + "library=python-storage,"
+                + "api={},".format(res.get("ApiName"))
+                + "op={},".format(res.get("Op"))
+                + "workers={},".format(workers)
+                + "object_size={},".format(object_size)
+                + "transfer_offset={},".format(res.get("TransferOffset", 0))
+                + "transfer_size={},".format(res.get("TransferSize", object_size))
+                + "app_buffer_size={},".format(res.get("AppBufferSize"))
+                + "chunksize={},".format(res.get("TransferSize", object_size))
+                + "crc32c_enabled={},".format(res.get("Crc32cEnabled"))
+                + "md5_enabled={},".format(res.get("MD5Enabled"))
+                + "cpu_time_us={},".format(res.get("CpuTimeUs"))
+                + "peer='',"
+                + f"bucket_name={bucket_name},"
+                + "retry_count='',"
+                + f"status={status}"
+                + "}"
+                f"{throughput}"
+            )
+
+            print(cloud_monitoring_output)
+
+
+def cleanup_directory_tree(directory):
+    """Clean up directory tree on disk."""
+    try:
+        shutil.rmtree(directory)
+    except Exception as e:
+        logging.exception(f"Caught an exception while deleting local directory\n {e}")
+    logging.info("Successfully removed local directory")
```

Comment:
Won't this result in an INFO-level log of success even if the attempt is a failure as per the exception caught above?

Reply:
good catch; updated!